src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -5200,6 +5541,39 @@ private static XmlNode[] FindNodes(object obj, string propertyName, StringCompar
 
             return retValue.ToArray();
         }
+
+        private static XmlNode FindFirstNodeOrDefault(object obj, MemberNamePredicate predicate)
```

Comment:
Actually, this might be refactored into a `HasProperty`. The formatting system needs to determine if an object has any property that is not a remoting property. That has previously been done in a very expensive way. When this code was written, I used this method in a place where I needed to find the NoteProperty that determined the output stream, but since that is now refactored, the question we want to have answered is if the psobject has any property that's not one of the magic remoting properties.

Reply:
But that is the reason why I want to stop with the first property. I want do to as little work as possible, and stop at the first match. 

But I guess the right thing to do is change it into a boolean test.

Reply:
Renaming...